File: frontend/src/lib/api/fleet.ts
import { client } from './client';
import { ENDPOINTS } from '../constants/api';
import type { DriverProfile } from '../types/api';

// Строгий контракт для створення сутності водія флоту
export interface CreateDriverRequest {
	license_number: string;
	phone?: string;
}

export const fleetAPI = {
	createProfile: (data: CreateDriverRequest) =>
		client.post<DriverProfile>(ENDPOINTS.FLEET.DRIVERS, data),
	getProfile: () => client.get<DriverProfile>(ENDPOINTS.FLEET.DRIVERS)
};

File: backend/internal/router/fleet.go
package router

import (
	"net/http"
	fleetapi "github.com/SiracencoSerghei/devtrack-app/backend/internal/contexts/fleet/api"
)

const fleetBase = "/api/fleet"

func registerFleetRoutes(mux *http.ServeMux, authMw func(http.Handler) http.Handler, h fleetapi.Handler) {
	mux.Handle("POST "+fleetBase+"/drivers", authMw(http.HandlerFunc(h.CreateProfile)))
	mux.Handle("GET "+fleetBase+"/drivers", authMw(http.HandlerFunc(h.GetProfile)))
}